File: frontend/src/api/movies.js
const BASE_URL = import.meta.env.VITE_API_BASE_URL

export const searchMovies = async (query) => {
  const res = await fetch(`${BASE_URL}/api/v1/movies/?search=${query}&limit=20`)
  const data = await res.json()
  return data.movies
}

export const getMovies = async (page = 1, genre = null) => {
  let url = `${BASE_URL}/api/v1/movies/?page=${page}&limit=20`
  if (genre) url += `&genre=${genre}`
  const res = await fetch(url)
  const data = await res.json()
  return data.movies
}

export const getMovie = async (id) => {
  const res = await fetch(`${BASE_URL}/api/v1/movies/${id}`)
  return await res.json()
}

File: backend/app/api/v1/movies.py
from fastapi import APIRouter, Query, HTTPException
from app.core.database import supabase

router = APIRouter()

@router.get("/")
def get_movies(
    page: int = Query(1, ge=1),
    limit: int = Query(20, le=100),
    genre: str = Query(None),
    search: str = Query(None),
):
    offset = (page - 1) * limit
    query = supabase.table("movies").select("*")

    if search:
        query = query.ilike("title", f"%{search}%")

    if genre:
        query = query.ilike("genre", f"%{genre}%")

    result = query.range(offset, offset + limit - 1).execute()
    return {"movies": result.data, "page": page, "limit": limit}


@router.get("/{movie_id}")
def get_movie(movie_id: int):
    result = supabase.table("movies").select("*").eq("id", movie_id).single().execute()
    if not result.data:
        raise HTTPException(status_code=404, detail="Movie not found")
    return result.data


@router.get("/search/titles")
def search_movies(q: str = Query(..., min_length=1)):
    result = supabase.table("movies").select("id, title, year, genre, imdb_rating, poster_url").ilike("title", f"%{q}%").limit(10).execute()
    return {"results": result.data}